x-pack/plugins/maps/public/store/map.js

```diff
@@ -128,13 +128,29 @@ export function map(state = INITIAL_STATE, action) {
         ...state,
         goto: null,
       };
+    case SET_LAYER_ERROR_STATUS:
+      const { layerList } = state;
+      const layerIdx = getLayerIndex(layerList, action.layerId);
+      if (layerIdx === -1) {
+        return state;
+      }
+
+      return {
+        ...state,
+        layerList: [
+          ...layerList.slice(0, layerIdx),
+          {
+            ...layerList[layerIdx],
+            __isInErrorState: true,
+            __errorMessage: action.errorMessage
+          },
+          ...layerList.slice(layerIdx + 1)
+        ]
+      };
```

Comment:
I'm wondering if it's time we revisit the `updateLayerInList` and update it to accept multiple attributes so we don't have to repeat this chunk of logic in the reducer